### PR TITLE
remove unused definitions, params, args from libs-scala and test tool Scala code

### PR DESCRIPTION
**Pull request by S11001001**
As found by `-Ywarn-unused` in #6907.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala [hunk 1]
```diff
@@ -50,9 +50,7 @@ object Assertions extends DiffExtensions {
         }
       // None both represents pattern that we do not care about as well as
       // exceptions that have no message.
-      case (GrpcException(GrpcStatus(`expectedCode`, Some(msg)), _), None) => ()
-      case (GrpcException(GrpcStatus(`expectedCode`, None), _), None) =>
-        ()
+      case (GrpcException(GrpcStatus(`expectedCode`, Some(_) | None), _), None) => ()
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala:53]: Isn't `Some(_) | None` the same as `_`?

```suggestion
      case (GrpcException(GrpcStatus(`expectedCode`, _), _), None) => ()
```

File: ledger/test-common/src/main/scala/com/digitalasset/platform/testing/TimeBoundObserver.scala [hunk 2]
```diff
@@ -30,7 +30,8 @@ final class TimeBoundObserver[T](duration: FiniteDuration)(
   }
 
   override def onCompleted(): Unit = {
-    val _succeeded = promise.trySuccess(buffer.result())
-    val _cancelled = Context.current().withCancellation().cancel(null)
+    promise.trySuccess(buffer.result())
+    Context.current().withCancellation().cancel(null)
+    ()
```
Comment (SamirTalwar-DA) [inline ledger/test-common/src/main/scala/com/digitalasset/platform/testing/TimeBoundObserver.scala:35]: Much prettier, thanks!

File: libs-scala/resources/src/main/scala/com/digitalasset/resources/ResourceOwner.scala [hunk 3]
```diff
@@ -41,7 +41,7 @@ trait ResourceOwner[+A] {
   }
 
   /** @see [[Resource.withFilter()]] */
-  def withFilter(p: A => Boolean)(implicit executionContext: ExecutionContext): ResourceOwner[A] =
+  def withFilter(p: A => Boolean): ResourceOwner[A] =
```
Comment (SamirTalwar-DA) [inline libs-scala/resources/src/main/scala/com/digitalasset/resources/ResourceOwner.scala:44]: Good catch. This was dangerous.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala (hunk 1 above)
Comment (S11001001) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala:53]: `null` matches neither 🙂 

Strictly speaking it doesn't preserve semantics, but it is probably what we meant anyway so I'll adopt that change.

Comment (SamirTalwar-DA) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala]: You are technically correct. The best kind of correct.

**Pull request merged**